client/sns/v2/publisher.go

```diff
@@ -67,9 +67,13 @@ func (p Publisher) Publish(ctx context.Context, input *sns.PublishInput) (messag
 	}
 
 	start := time.Now()
-
 	out, err := p.api.PublishWithContext(ctx, input)
-	observePublish(span, start, *input.TopicArn, err)
+	if input.TopicArn != nil {
```

Comment:
As described in [issue 387](https://github.com/beatlabs/patron/issues/387) this protection corrects the behavior of the `examples/amqp/main.go`. It also makes the service more robust against panicing